### PR TITLE
[VEN - 820] Optimize Comptroller

### PR DESCRIPTION
**Pull request by defcon022**
## Description

Before including the updated functionality of the XVS speeds, Comptroller size was 23.906KB (develop branch).

After including the XVS speeds functionality, size gone up to 25.1KB.

Refactored some code and bring back the Comptroller size to 24.503KB, still which was not enough.

XVS functionality contains a function([_upgradeSplitVenusRewards](https://github.com/VenusProtocol/venus-protocol/pull/181/files#diff-e4af81e44672d917e01c9b3ab9df9042fa6b35d1c02cb3b54cfca095552abcbdR1131)) which would be run once to update the venusSpeeds to venusSupplySpeeds and venusBorrowSpeeds for the existing markets speeds.

Plan to deploy the Comptroller:
- To deploy the XVS functionality we removed Verify hooks from the comptroller which bring down the size to 23.8KB. and execute [_upgradeSplitVenusRewards](https://github.com/VenusProtocol/venus-protocol/pull/181/files#diff-e4af81e44672d917e01c9b3ab9df9042fa6b35d1c02cb3b54cfca095552abcbdR1131)() and set the speeds.
- Deploy a Comptroller with the Verify hooks and remove _upgradeSplitVenusRewards function from it which would bring down the  size of the Comptroller to 23.9kb.

Note:- Comptroller without the hooks are added in file [UpdatedComptroller.sol](https://github.com/VenusProtocol/venus-protocol/blob/5003d7d22d2aa9852fba474ed84c10bd4ce297a7/contracts/Comptroller/UpdatedComptroller.sol). And the Comptroller with full functionality is at same place [Comptroller.sol](https://github.com/VenusProtocol/venus-protocol/blob/5003d7d22d2aa9852fba474ed84c10bd4ce297a7/contracts/Comptroller/Comptroller.sol) to avoid confusion.

## Checklist

- [ ] I have updated the documentation to account for the changes in the code.
- [ ] If I added new functionality, I added tests covering it.
- [ ] If I fixed a bug, I added a test preventing this bug from silently reappearing again.
- [ ] My contribution follows [Venus contribution guidelines](docs/CONTRIBUTING.md).

### CODE REVIEW

File: contracts/Comptroller/Comptroller.sol [hunk 1]
```diff
@@ -321,13 +318,7 @@ contract Comptroller is ComptrollerV9Storage, ComptrollerInterfaceG2, Comptrolle
      * @param actualMintAmount The amount of the underlying asset being minted
      * @param mintTokens The number of tokens being minted
      */
-    function mintVerify(address vToken, address minter, uint actualMintAmount, uint mintTokens) external {
-        // Shh - currently unused
-        vToken;
-        minter;
-        actualMintAmount;
-        mintTokens;
```
[the remark this answers is not in the document]
Comment (kkirka) [reply contracts/Comptroller/Comptroller.sol:329]: I agree these "empty statements" are confusing – the auditors constantly report unused code, so I'd be happy if we get rid of them. On the other hand, I don't know how we can force-silince the warnings in Solidity without using transpilers.

File: contracts/Comptroller/Comptroller.sol [hunk 2]
```diff
@@ -1424,17 +1372,25 @@ contract Comptroller is ComptrollerV9Storage, ComptrollerInterfaceG2, Comptrolle
...
 
-        if (amount == 0 || amount > venusRemaining) {
+        if (amount == 0 || amount > xvs.balanceOf(address(this))) {
             return amount;
         }
 
-        // If user's not bankrupt, user can get the reward,
-        // so the liquidators will have chances to liquidate bankrupt accounts
-        if (!bankrupt) {
+        if (!(shortfall > 0)) {
```
[the remark this answers is not in the document]
Comment (defcon022) [reply contracts/Comptroller/Comptroller.sol:1393]: Fixed.

File: .env.example [hunk 3]
```diff
@@ -0,0 +1,2 @@
+FORK_MAINNET=false
+QUICK_NODE_KEY=
```
Comment (coreyar) [inline .env.example:2]: This should be a URL so that we can easily change providers.

File: contracts/Comptroller/Comptroller.sol [hunk 4]
```diff
@@ -49,8 +49,11 @@ contract Comptroller is ComptrollerV9Storage, ComptrollerInterfaceG2, Comptrolle
     /// @notice Emitted when Venus VAI Vault rate is changed
     event NewVenusVAIVaultRate(uint oldVenusVAIVaultRate, uint newVenusVAIVaultRate);
 
-    /// @notice Emitted when a new Venus speed is calculated for a market
-    event VenusSpeedUpdated(VToken indexed vToken, uint newSpeed);
+    /// @notice Emitted when a new borrow-side XVS speed is calculated for a market
```
Comment (coreyar) [inline contracts/Comptroller/Comptroller.sol:52]: Let's update the natspec for with these events per our agreed upon convention

File: contracts/Comptroller/UpdatedComptrollerInterface.sol [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    function venusBorrowerIndex(address, address) external view returns (uint);
+    function venusBorrowState(address) external view returns (uint224, uint32);
+    function venusSupplyState(address) external view returns (uint224, uint32);
+}
+
+// interface IVAIVault {
+//     function updatePendingRewards() external;
+// }
+
+// interface IComptroller {
```
Comment (coreyar) [inline contracts/Comptroller/UpdatedComptrollerInterface.sol:80]: Remove commented code

File: contracts/Comptroller/Comptroller.sol [hunk 6]
```diff
@@ -1423,18 +1373,14 @@ contract Comptroller is ComptrollerV9Storage, ComptrollerInterfaceG2, Comptrolle
             && user != 0x24e77E5b74B30b026E9996e4bc3329c881e24968,
             "Blacklisted"
         );
-
+
```
Comment (kkirka) [inline contracts/Comptroller/Comptroller.sol:1376]:
```suggestion

```

**Review by kkirka — APPROVED** [review]

File: contracts/Comptroller/Comptroller.sol (hunk 6 above)
Comment (coreyar) [reply contracts/Comptroller/Comptroller.sol:1376]: We could leave this for now. I'm going to add solidity linting as soon as this and 185 and 140 are merged

**Pull request merged**